docs/DeveloperGuide.md

```diff
@@ -0,0 +1,119 @@
+
+1. Go to `File` > `Settings…`​ (Windows/Linux), or `IntelliJ IDEA` > `Preferences…`​ (macOS).
+2. Select `Editor` > `Code Style` > `Java`.
+3. Click on the `Imports` tab to set the order
+   * For `Class count to use import with '*'` and `Names count to use static import with '*'`: Set to `999` to prevent IntelliJ from contracting the import statements
+   * For `Import Layout`: The order is `import static all other imports`, `import java.*`, `import javax.*`, `import org.*`, `import com.*`, `import all other imports`. Add a ``<blank line>`` between each `import`.
+
+Optionally, you can follow the [Using Checkstyle](UsingCheckstyle.md) document to configure *Intellij* to check-style compliance as you write code.
+
+### Before writing code
+1. Do check out our [process guide](../docs/Process.md) before submitting any PR with your changes.
+
+## Architecture
+
+ ![architecture](images/architecture.png)
+*Figure 1. Overall architecture of RepoSense*
+
+### Parser
+`Parser` contains two classes:
+ * [`ArgsParser`](/src/main/java/reposense/parser/ArgsParser.java): Parses the user-supplied command line arguments into a `CliArguments` object.
+ * [`CsvParser`](/src/main/java/reposense/parser/CsvParser.java): Parses the the user-supplied CSV config file and produces a list of `RepoConfiguration` for each repository for analysis.
+
+
+### Git
+`Git` contains the wrapper classes for respective *git* commands.
+ * [`GitDownloader`](/src/main/java/reposense/git/GitDownloader.java): Wrapper class for `git clone` functionality. Clones the repository from *GitHub* into a temporary folder in order to run the analysis.
+ * [`GitChecker`](/src/main/java/reposense/git/GitChecker.java): Wrapper class for `git checkout` functionality. Checks out the repository by branch name or commit hash.
+
+
+### AuthorshipReporter
+[`AuthorshipReporter`](/src/main/java/reposense/authorship/AuthorshipReporter.java),
+ 1. Uses [`FileInfoExtractor`](/src/main/java/reposense/authorship/FileInfoExtractor.java) to traverse the repository to find all relevant files.
+ 2. Generates a [`FileInfo`](/src/main/java/reposense/authorship/model/FileInfo.java) for each relevant file, which contains the path to the file and a list of [`LineInfo`](/src/main/java/reposense/authorship/model/LineInfo.java) representing each line of the file.
+ 3. Uses [`FileInfoAnalyzer`](/src/main/java/reposense/authorship/FileInfoAnalyzer.java) to analyze each file using `git blame` or annonations to set the `Author` for each `LineInfo`.
+ 4. Generates a [`FileResult`](/src/main/java/reposense/authorship/model/FileResult.java) which consolidates the authorship results into a *Map* of each author's line contribution to the file.
+ 5. Uses [`FileResultAggregator`](/src/main/java/reposense/authorship/FileResultAggregator.java) to aggregate all `FileResult` into an [`AuthorshipSummary`](/src/main/java/reposense/authorship/model/AuthorshipSummary.java).
+
+
+### CommitsReporter
```

Comment:
Perhaps put the CommitsReporter above Authorship? To follow the actual program flow.